src/main/java/org/shanerx/tradeshop/commands/CommandCaller.java

```diff
@@ -53,14 +53,23 @@ public boolean onCommand(CommandSender sender, Command cmd, String label, String
         cmdPass = new CommandPass(sender, cmd, label, args);
         command = Commands.getType(cmdPass.getArgAt(0));
 
+
         if (!cmdPass.hasArgs()) {
             sender.sendMessage(Message.INVALID_ARGUMENTS.getPrefixed());
             return true;
 
         }
 
         if (command == null) {
-            sender.sendMessage(Message.INVALID_ARGUMENTS.getPrefixed());
+	        CustomCommandHandler handler = CustomCommandHandler.getInstance();
+	        String subcmd = cmdPass.getArgAt(0);
+
+	        if (!handler.isAvailable(subcmd)) {
```

Comment:
Either isAvailable is returning backwards or this shouldn't be !

Reply:
isAvailable returns true if that subcommand has not been registered. Hence, if the value is false, it means the command has already been registered by some addon or by the plugin itself (aka it is one of the entries of the enum Commands). Clearly, the second option is not possible as if it were, the null check would fail.

Reply:
ah so is available to register, got it, makes more sense now